date_range/models/date_range.py

```diff
@@ -28,11 +28,27 @@ def _default_company(self):
     active = fields.Boolean(
         help="The active field allows you to hide the date range without "
         "removing it.", default=True)
+    parent_id = fields.Many2one(
+        comodel_name='date.range', string="Parent", index=1)
 
     _sql_constraints = [
         ('date_range_uniq', 'unique (name,type_id, company_id)',
          'A date range must be unique per company !')]
 
+    @api.constrains('parent_id', 'date_start', 'date_end')
+    def _validate_child_range(self):
+        for this in self:
+            if not this.parent_id:
+                continue
+            child_range = (
+                this.parent_id.date_start <= this.date_start and
+                this.parent_id.date_end >= this.date_end
+            )
+            if not child_range:
+                raise ValidationError(
+                    _("%s can not be child of %s") % (
```

Comment:
In your messages you should always be explicit. Otherwise the user will wonder what is wrong. Tell that the new range has a start date before, or an end date after the start of the parent range.

Reply:
:+1: with @NL66278